### PR TITLE
VEX-in-Wolfi documentation

### PR DESCRIPTION
**Pull request by puerco**
This PR adds the first document detailing how VEX is originated and flows from wolfi config files all the way down to security scanners.

/cc @ltagliaferri @sheesh (for possible sync with academy)

Signed-off-by: Adolfo García Veytia (Puerco) <puerco@chainguard.dev>

### CODE REVIEW

**Comment by lisa** [comment]
> This PR adds the first document detailing how VEX is originated and flows from wolfi config files all the way down to security scanners.
> 
> /cc @lisa @sheesh (for possible sync with academy)
> 
> Signed-off-by: Adolfo García Veytia (Puerco) [puerco@chainguard.dev](mailto:puerco@chainguard.dev)

# `@lisa` User Notification

Hello! You're receiving this notice because you've tagged me, @lisa, on GitHub, likely in error. Lisa is a common first name, and happens to be mine. I use the @lisa username here on GitHub. While being tagged in various repositories is an interesting way for me to discover new projects, it's likely not your intention. Did you mean to tag someone else? Did you mean to use `@lisa` instead, in a [word as word](https://en.wikipedia.org/wiki/Use%E2%80%93mention_distinction) usage?

# Next Steps

I'm unsubscribing from this issue, pull request or project so if you meant to tag someone else to notify them, I would recommend that. If you meant to use my user ID as a plain string, without having GitHub notify me, you can do so by wrapping with backticks (the \` character), like so: `@lisa`.

File: docs/vex.md [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+of supply chain metadata as part of its build processes. All WolfiOS packages
+get a software bill of materials (SBOM) and to complement it, Vulnerability
+Exploitability Exchange (VEX) metadata can be generated to help with false
+positives in security scans exacerbated by the extreme transparency offered
+by all the available data.
+
+If you are unfamiliar with VEX, you can think of it as a negative security
+advisory. A document where software authors inform users that a vulnerability
+does not affect a program. For more information check out the VEX resources
+at Chainguard academy.
```
Comment (luhring) [inline docs/vex.md:13]: This would be a good place for one or more links to more thorough explanations of VEX.

File: docs/vex.md [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+Security advisories contain the patch history of the package. The data in the
+advisories lets users know when the project became aware of a vulnerability, if
+it affects the package, which versions are impacted, and how quickly it was
+handled by the Wolfi team after disclosure. Sometimes security scanners will
+miss these patches and falsely report the package as vulnerable. Here is where
+VEX can help.
+
+## How Wolfi VEX Metadata is Generated
+
+By Translating the Wolfi advisories into VEX metadata, the project can piece
```
Comment (luhring) [inline docs/vex.md:34]: nit:
```suggestion
By translating the Wolfi advisories into VEX metadata, the project can piece
```

File: docs/vex.md [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+[Sigstore](https://sigstore.dev/)'s
+[cosign SBOM spec](https://github.com/sigstore/cosign/blob/main/specs/SBOM_SPEC.md).
+
+## A Real Life Example: Understanding a VEX Document
+
+Let's take a look at an example. We will enerate a VEX Document using the SBOM
+of [the Chainguard `static` base image](https://github.com/chainguard-images/images/tree/main/images/static)
+(the example uses the digest of the amd64 variant at the time of writing).
+
+First. let’s look at what is inside the SBOM. The SBOM is attached to the image
```
Comment (luhring) [inline docs/vex.md:139]: nit:
```suggestion
First, let’s look at what is inside the SBOM. The SBOM is attached to the image
```

File: docs/vex.md [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+[impact status](https://github.com/openvex/spec/blob/main/OPENVEX-SPEC.md#status-labels):
+“fixed”.
+
+The document also specifies the subcomponent where the vulnerability originated:
+the Wolfi package identified by the purl `pkg:apk/wolfi/glibc-locale-posix@2.36-r4`.
+
+When evaluating the `static` image, a scanner can read the VEX metadata in the document 
+and turn off any false positives it finds by relying on data originated at
+the source: the Wolfi maintainers themselves. While scanners implement native
+filtering, [OpenVEX's `vexctl`](https://github.com/openvex/vexctl) can be used
```
Comment (luhring) [inline docs/vex.md:231]: >While scanners implement native filtering

I think this phrase might be able to be misinterpreted as claiming that scanners implement native filtering _today_. I'd suggest something like:

>While scanners are preparing to support VEX filtering natively, ...

**Review by luhring — COMMENTED** [review]
I really like this! 😍 

I have a few overall comments:

1. This explanation doc should probably go right **in the Wolfi repo**, not here in the `wolfictl` tool repo. It's describing a core philosophy and approach for Wolfi security.

2. Unless @kaniini has a different take, I'd suggest striking all "WolfiOS" appearances and always just saying "Wolfi". I think being consistent will make this more clear.

3. We should always capitalize "VEX": 1) to make it easier for readers to follow, and 2) to distinguish from the English (and unrelated) word "vex".

**Comment by kaniini** [comment]
What @luhring said is right.  It's always "Wolfi", and never "Wolfi OS" or "WolfiOS".  The "os" repository is just one deliverable of the Wolfi project (and is the main one, of course).

File: docs/vex.md (hunk 1 above)
Comment (puerco) [reply docs/vex.md:13]: I was planning to link to Academy, but I changed it to our blog post for now.

**Comment by puerco** [comment]
Cool, I was trying to somehow differentiate between wolfi the OS and wolfi the project. But I actually don't like WolfiOS at all.

I've addressed the nits, should I close this PR and reopen it somewhere in `os`? What do you think @kaniini and @luhring ?

**Comment by luhring** [comment]
> I've addressed the nits, should I close this PR and reopen it somewhere in `os`? What do you think @kaniini and @luhring ?

Awesome! Yeah, that'd be my suggestion. "Where in `os`?" is a good question, since the root of the repo is a bit much. Maybe a `docs` folder for now? And at some point we could move the "How to Patch..." markdown file there, and others, too.

**Comment by puerco** [comment]
Closing this PR as `wolfictl vex` was removed.
/cc @jspeed-meyers

**Pull request closed without merging**